### DISCUSSION TITLE
Feature request: allow filter pending logs from `publicClient.watchEvent`

### DISCUSSION THREAD

**Discussion by shirotech** [wevm/viem#1556] [Idea / Feature Request]
For example from ethers, we can do:

```ts
await provider.on(
  {fromBlock: -2, topics: ['0xd78ad95fa46c994b6551d0da85fc275fe613ce37657fb8d5e3d130840159d822']},
  async (log) => {
    console.log(log);
  },
)
```

Right now there's no way to do this on `viem` as the `fromBlock` prop is not available. Getting an error:
<img width="926" alt="Screenshot 2023-12-05 at 6 55 27 am" src="https://github.com/wevm/viem/assets/1149825/5f61fa70-b075-4cd3-a54b-652817f99c31">

**Comment by dsldsl** [comment]
Also would find this useful
